src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -3640,6 +3640,90 @@ def groupby_shift(
             )
         )
 
+    def groupby_size(
+        self,
+        by: Any,
+        axis: int,
+        groupby_kwargs: dict[str, Any],
+        agg_args: tuple[Any],
+        agg_kwargs: dict[str, Any],
+        drop: bool = False,
+        **kwargs: dict[str, Any],
+    ) -> "SnowflakeQueryCompiler":
+        """
+        compute groupby with size.
+        With a dataframe created with following:
+        import pandas as pd
+
+        data = [[1,2,3], [1, 5, 6], [2, 5, 8], [2, 6, 9]]
+
+        df = pd.DataFrame(data, columns=["a", "b", "c"], index = ["tuna", "salmon", "catfish", "goldfish"])
+
+        df
+
+                  a  b  c
+
+        tuna      1  2  3
+        salmon    1  5  6
+        catfish   2  5  8
+        goldfish  2  6  9
+
+        df.groupby("a").size()
+
+        a
+        1    2
+        2    2
+        dtype: int64
+
+
+        Args:
+            by: mapping, series, callable, label, pd.Grouper, BaseQueryCompiler, list of such.
+                Use this to determine the groups.
+            axis: 0 (index) or 1 (columns).
+            groupby_kwargs: dict
+                keyword arguments passed for the groupby.
+            agg_args: tuple
+                The aggregation args, unused in `groupby_size`.
+            agg_kwargs: dict
+                The aggregation keyword args, unused in `groupby_size`.
+            drop: bool
+                Drop the `by` column, unused in `groupby_size`.
+        Returns:
+            SnowflakeQueryCompiler: The result of groupby_size()
+        """
+        level = groupby_kwargs.get("level", None)
+        is_supported = check_is_groupby_supported_by_snowflake(by, level, axis)
+
+        if not is_supported:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas GroupBy.size does not yet support pd.Grouper, axis == 1, by != None and level != None, by containing any non-pandas hashable labels, or unsupported aggregation parameters."
+            )
+        if not is_list_like(by):
+            by = [by]
+
+        positions_col_name = "__TEMP_POS_NAME__"
+        # We reset index twice to ensure we perform the count aggregation on the row
+        # positions (which cannot be null).
+        result = (
+            self.reset_index(drop=True)
+            .reset_index(drop=False, names=positions_col_name)
+            .take_2d_labels(slice(None), [positions_col_name] + by)
+            .groupby_agg(
+                by,
+                "count",
+                axis,
+                groupby_kwargs,
+                (),
+                {},
+            )
+        )
+        if not groupby_kwargs.get("as_index", True):
```

Comment:
if you simply pass the correct name to reset_index, will we need to do rename here? or there are some other situtation we are worried about?

Reply:
We need the new column name to be unique (there may be a `"size"` or a `MODIN_UNNAMED_SERIES_LABEL` named column already in the data). I'll add a comment there.

Reply:
if that is the case, it is also possible that we will have conflict with positions_col_name, right? then may be generate the column name by deconflict among the existing pandas labels with a random suffix, we can add a simple utility function under utils.py, we have one for quoted_identifiier generation generate_snowflake_quoted_identifiers_helper, but this one doesn't require any quotes there, might be easy to just get another one to operate at pandas label level

Reply:
or another simple way is that you can simply unquote the identifier of the index column, which is guaranteed to be unique among the column, but may not be that obvious on purpose when read the code

Reply:
>if that is the case, it is also possible that we will have conflict with positions_col_name, right?

Not really, unless the customer has a column named "__TEMP_POS_NAME__". I think this is fine as is, and if we want to refine the unique logic we should do it in a subsequent PR to unblock.

Reply:
I would at least append a random suffix to be safe, you do not know what user will do, sometimes it is not the name they add, it could be caused due to our backend processing where we are quoting and unquoting the identifiers. And can you create a ticket to follow this up?

Reply:
SNOW-1478730

Reply:
I also added a random 6 digit hex code at the end of the temporary name. I'll mark this as resolved and please unresolve it if there's something else that needs to be done.